PURL-TYPES.rst

```diff
@@ -221,6 +207,34 @@ nuget
 
       pkg:nuget/EnterpriseLibrary.Common@6.0.1304
 
+oci-artifact
+------------
+``oci-artifact`` for all artifacts stored in OCI Distribution-spec conformant
+registries including container images built by docker and others:
+
+- There is no canonical package repository for OCI artifacts. Therefore
+  ``oci-artifact`` pURLs must be registry agnostic.
+- OCI pURLs do not require a ``namespace``. The ``namespace`` is specific
+  to the physical location of the package.
+- The ``name`` is not case sensitive and must be lowercased. The name is the
+  last fragment of the repository name. For example if the repository
+  name is ``library/debian`` then the ``name`` is ``debian``.
+- The ``version`` is the ``@sha256:digest`` of the artifact and is strongly
+  suggested for oci artifacts.
+- Optional qualifiers may include:
+
+  - ``arch``: key for a package architecture, when relevant
+  - ``repository``: A repository URL where the artifact may be found, but not intended as the only location
+  - ``tag``: artifact tag that may have been associated with the digest at the time
+  - ``type``: type of artifact stored in an OCI registry
```

Comment:
Is this always a Media Type?

Reply:
Yes, will always be a media type. I can clarify in the description.

Reply:
It's a little more nuanced. We [initially used media types](https://github.com/opencontainers/artifacts/blob/master/artifact-authors.md#defining-a-unique-artifact-type) to squeeze into the existing `manifest.config.mediatype` format.
Moving forward, we use [artifactType](https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md#oras-artifact-manifest-properties) and have decoupled from `manifest.config.mediaType`. 
However, to avoid artifact authors from having to maintain two types, we are using iana.org mediaTypes to uniquely differentiate types. 
> For details on creating a unique artifactType, see [OCI Artifact Authors Guidance](https://github.com/opencontainers/artifacts/blob/master/artifact-authors.md)

Reply:
Using `type` here is confusing, given that purl already defines a `type`.

This is also ambiguous.

Does `type` refer to the media type of the artifact, the config mediaType (as [defined in OCI artifacts](https://github.com/opencontainers/artifacts/blob/master/artifact-authors.md#defining-oci-artifact-types)), or the `artifactType` from [oras artifacts](https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md#oras-artifact-manifest-properties)?

If the first, I'd suggest renaming to `mediaType`.

If the second, `artifactType` seems fine.

If the third, I think the purl `type` should probably be renamed to `oras-artifact`, given that they don't have a compatible schema with OCI Artifacts.

Reply:
> Using `type` here is confusing, given that purl already defines a `type`.

type is used by other purl types (see: maven) so I don't think this is a conflict with the purl spec but I don't have issues renaming it to `mediaType`.

Reply:
> type is used by other purl types (see: maven)

Oh cool, that's fine with me then. Can you clarify if this refers to the artifact's [media type](https://github.com/opencontainers/image-spec/blob/main/media-types.md) or its [artifact type](https://github.com/opencontainers/artifacts/blob/master/artifact-authors.md#defining-oci-artifact-types)?

Reply:
It should refer to the artifact's artifact type as defined [here](https://github.com/opencontainers/artifacts/blob/master/artifact-authors.md#defining-oci-artifact-types). This is illustrated in some of the examples above as well.

Reply:
In that case, I don't think this PR is about OCI artifacts in general, but specifically about OCI Artifacts, which wouldn't completely encompass container images.

Reply:
> Oh cool, that's fine with me then. Can you clarify if this refers to the artifact's media type or its artifact type?

@jonjohnsonjr, this was also your feedback to use the same format for both `manifest.config.mediaType` and `oras.manifest.artifactType`

SInce mediaType has such broad meanings, we were using artifactType, which happens to use the iana mediaType format.

Reply:
If this is the `artifactType` field from an OCI Artifact, then are we saying this purl does not apply to OCI Images (and docker images) since those do not have an `artifactType` field? Or is there a decision tree based on some logic that maps the current manifest media type + config media type to this `type` field?

Reply:
> If this is the `artifactType` field from an OCI Artifact
 
The `artifactType` field is from ORAS Artifacts, not OCI Artifacts.